### DISCUSSION TITLE
Nuxt generate - multiple dynamic routes each one with several variations

### DISCUSSION THREAD

**Discussion by Jesus82** [nuxt/nuxt#9746] [Questions (Nuxt 2)]
Let's say that I want to generate multiple dynamic routes. In this example I want to generate one with `users` and one with `posts`. Then with each user and with each post I want also to render different language variations. For this, I will generate a route with a different preffix (`/posts/...` for english, `/es/posts/`... for spanish, `/pt/posts/`... for portuguese).

Based in [this issue](https://github.com/nuxt/nuxt.js/issues/1018), I came up with this, but it doesn't seem to work (it doesn't generate any of the dynamic routes). Any advice would be welcome.

    routes() {
      let endPoint = "https://my.endpoint/"
      let languagesArray = ['', '/es', '/pt']
      let brewers = axios.get(endPoint + 'users').then((res) => {
        return res.data.map((item) => {
          let mapArray = []
          for (var i = 0; i > languagesArray.length; i++ ) {
            mapArray.push({
              route: languagesArray[i] + '/user/' + item.slug + '/',
              payload: item
            })
          }
          return mapArray
        })
      })
      let posts = axios.get(endPoint + 'posts').then((res) => {
        return res.data.map((item) => {
          let mapArray = []
          for (var i = 0; i > languagesArray.length; i++ ) {
            mapArray.push({
              route: languagesArray[i] + '/post/' + item.slug + '/',
              payload: item
            })
          }
          return mapArray
        })
      })
      return Promise.all([users, posts]).then(values => {
        return [...values[0], ...values[1]]
      })
    }

**Comment by ghost** [comment]
I have the same issue. I do this and it's perfectly working:
[https://medium.com/js-dojo/how-i-generated-dynamic-routes-for-different-nuxt-js-pages-ce2ee6972743](url)

hope it helps.
